runtime/caml/signals.h

```diff
@@ -37,10 +37,14 @@ CAMLextern void caml_process_pending_actions (void);
    finalisers, and Memprof callbacks. Assumes that the runtime lock is
    held. Can raise exceptions asynchronously into OCaml code. */
 
+/* Same as [caml_process_pending_actions], but returns the reified
+   result instead of raising exceptions directly (if any). */
+CAMLextern caml_result caml_process_pending_actions_result (void);
+
+/* Returns [Val_unit] or an encoded exception.
+   Superseded by the safer [_result] variant above,
+   kept around for compatibility. */
 CAMLextern value caml_process_pending_actions_exn (void);
```

Comment:
You can move the superseded functions towards the end of the public interface, and have the documentation simply say `/* Superseded by …_result */`